lib/src/builder/parsing/declarations_from_members.dart

```diff
@@ -209,6 +212,12 @@ Iterable<BoilerplateDeclaration> getBoilerplateDeclarations(
     ..sort((a, b) => b.versionConfidences.maxConfidence.confidence
         .compareTo(a.versionConfidences.maxConfidence.confidence));
   for (final factory in factoriesMostToLeastConfidence) {
+    if (factory.isFunctionComponentFactory) {
+      _functionComponentFactories.add(factory);
+      // will be validated below the for-loop.
```

Comment:
Is there a way we could combine the below for-loop into this one? It would be nice if all of the grouping logic was in one place, and make it easier to tell what grouping logic takes precedence.

I had some old logic that grouped factories together: https://github.com/Workiva/over_react/commit/f62af3fb005531b3576d9fd707db52ff9e52a7c8

So maybe parts of that could be used here. M Perhaps something like

```diff    

-  final factoriesMostToLeastConfidence = List.of(members.factories)
+  final factoriesMostToLeastConfidence = factoryGroups.map((group) => group.bestFactory).toList()
     ..sort((a, b) => b.versionConfidences.maxConfidence.confidence
         .compareTo(a.versionConfidences.maxConfidence.confidence));
   for (final factory in factoriesMostToLeastConfidence) {
...

     final component = getComponentFor(factory, members.components);
     if (component != null && hasBeenConsumed(component)) {
       continue;
     }

+    if (factory.isFunctionComponentFactory) {
+      final factories = factoryGroups.firstWhere((group) => group.bestFactory == factory);
+      yield PropsMapViewOrFunctionComponentDeclaration(factories: factories)
+    }
```

Reply:
Updated it! That factory grouping is really nice and it fixed some of the other parsing problems I was having too!